### PR TITLE
Backburner 2.0 re-write initial efforts

### PR DESCRIPTION
**Pull request by lynchbomb**
Updates include:

- [x] migrating all "lib" files to TypeScript. 
- [x] addressing issues with rollup causing our test suite/failing tests
- [x] dependency updates/cleanup package.json
- [x] version bump to 0.4.0
- [x] TypeScript config changes within ember-cli-build

[edit: fixed checklist]

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,4 +16,4 @@ env:
     - secure: Sx7SVc3tFbpTdtg494J32I9YDxedYJtDEifssN9Oe8zK02Jw+i74e45Njk1vr/REJjefNa26zMGI+w0UnSywRsKhogN4pzqtaDJKK8H8JaOFf5WoSXbSpEnXMoOLD9DzKjj0j1qxSqXnPlv4czyyO5firmYqOPPAziDED9v8eLM=
     - secure: Lo7dEbRFHNrQL29iUs88zFQyRNWcK7lnLIoPlBIUDiLOXamsUilxykAPmifIK34yFS4GqN+zv9mkcVEkieRQV5LlZgNd1RLJ5nDM6bkMSRM1K14K36EUy3ldU9Ld1d/sa5yx4cxJXqNRjfF/d+i7OTDQduLYmrFLhLPPu2HetbA=
     - secure: cAsSmFA77dBTgo7JHT3kcXlwAfilDoOwrVo3b77SHj64rWVUjB3xyaCWY+3gCoULTDD/fuEO80TWwzfV2sQGASvJyrSNLw3P7//Ym/dkHcTdQ3CfCagCf689/PfFTMM1+v6gZHSYcryJzRWkLolEgFlkU+lw4rwVlRjrKSwnRgQ=
-    - secure: bNBRMlxWFRbAR04OI1SfwfYJis9KTm0C6O/xkAuMiardqJ6z+EpTXKSplAu6MlUM5VsiOON2ZNwo3bwmtshWhGiUhOVnjZCBtuVcJXyj/RQk9ug8A8pFKYn4o3Yn57H6o4tj/Gw3/NV5ocbw5MzRbgCSFkLbHl9lPDRTYyX2SwM=
+    - secure: bNBRMlxWFRbAR04OI1SfwfYJis9KTm0C6O/xkAuMiardqJ6z+EpTXKSplAu6MlUM5VsiOON2ZNwo3bwmtshWhGiUhOVnjZCBtuVcJXyj/RQk9ug8A8pFKYn4o3Yn57H6o4tj/Gw3/NV5ocbw5MzRbgCSFkLbHl9lPDRTYyX2SwM=
```
Comment (krisselden) [inline .travis.yml]: what's removing the last \n?

Comment (lynchbomb) [inline .travis.yml]: Egh - looks like a setting within VSCode. I'll address.

File: npm-debug.log [hunk 2]
```diff
@@ -0,0 +1,9184 @@
+98049 info linkStuff es6-iterator@2.0.0
```
Comment (rwjblue) [inline npm-debug.log]: Should remove `npm-debug.log`...

Comment (lynchbomb) [inline npm-debug.log]: Removed and added to .gitignore

File: .travis.yml (hunk 1 above)
Comment (lynchbomb) [inline .travis.yml]: \n addressed. included .editorconfig and .vscode/settings.json.

File: ember-cli-build.js [hunk 3]
```diff
@@ -3,67 +3,49 @@
...
-const mkdirp = require('mkdirp').sync;
-
-class TestIndex extends Plugin {
-  constructor(input) {
-    super([input], {
-      persistentOutput: true,
-      annotation: 'tests/index.js'
-    });
+const typescript = require('broccoli-typescript-compiler');
+const ts = typescript(stew.find('lib'), {
```
Comment (stefanpenner) [inline ember-cli-build.js]: You can pass `lib` straight to `typescript`, let me also recommend using `__dirname + '/lib'` that way this broccoli pipeline can be used if `process.cwd()` is not the root of this project (if lets see ember is building)

File: ember-cli-build.js [hunk 4]
```diff
@@ -3,67 +3,49 @@
...
-    let outputPath = this.outputPath + '/tests';
-    let testModules = walkSync(inputPath, {
-      globs: ['**/*.js']
-    }).map(file => {
-      let moduleId = file.slice(0, -3);
-      return `import './${moduleId}';`;
-    }).join('\n');
-    mkdirp(outputPath);
-    fs.writeFileSync(outputPath + '/index.js', testModules);
+const ts = typescript(stew.find('lib'), {
```
Comment (stefanpenner) [inline ember-cli-build.js]: with this there there appears to be compile issue, specifically 1 ES6 export doesn't seem to be being compiled.  (which will need exploration).

Comment (lynchbomb) [inline ember-cli-build.js]: Updates above are committed. Appears that the 1 ES6 export not being compiled was/is a pre-existing condition on the current `master`. Needs additional exploration.

**Pull request merged**